homeassistant/components/telegram_bot/polling.py

```diff
@@ -35,26 +35,31 @@ def process_error(update: Update, context: CallbackContext) -> None:
 
 
 class PollBot(BaseTelegramBotEntity):
-    """Controls the Updater object that holds the bot and a dispatcher.
+    """Controls the Application object that holds the bot and an updater.
 
-    The dispatcher is set up by the super class to pass telegram updates to `self.handle_update`
+    The application is set up to pass telegram updates to `self.handle_update`
     """
 
     def __init__(self, hass, bot, config):
-        """Create Updater and Dispatcher before calling super()."""
+        """Create Application before calling super()."""
```

Comment:
There are no side effects in the parent init method. It just sets two attributes. I'd call super first since normally you want the child to be able to overwrite the parent attributes if needed.